### PR TITLE
feature/REST response for API

### PR DESCRIPTION
**Pull request by zabukovnikmaj**

### CODE REVIEW

File: Controllers/BaseController.php [hunk 1]
```diff
@@ -212,4 +212,46 @@ protected function getNameFromUuid(string $table, string $uuid): string
...
+     * @return void
+     */
+    public function apiData(): void
+    {
+        //TODO: adding authentication of the user, so only certain users can access this API
+
+        $table = $this->getFilenameFromClass();
+        $data = Storage::loadElements($table);
+
+        if (empty($data)) {
```
Comment (vojkovoga) [inline Controllers/BaseController.php:228]: That's not true. Storage can be empty.

File: Controllers/BaseController.php [hunk 2]
```diff
@@ -212,4 +212,46 @@ protected function getNameFromUuid(string $table, string $uuid): string
...
+    /**
+     * function for sending API response with specific status code and data
+     *
+     * @param int $statusCode
+     * @param array $data
+     * @return void
+     */
+    protected function sendResponse(int $statusCode, array $data): void
+    {
+        http_response_code($statusCode);
```
Comment (vojkovoga) [inline Controllers/BaseController.php:252]: You can use your `json` helper method

File: config/routes.php [hunk 3]
```diff
@@ -48,6 +48,17 @@
     ],
 
 
+    '/api/branchOffice/' => [
+        'GET' => [BranchOffice::class, 'apiData']
```
Comment (vojkovoga) [inline config/routes.php:52]: Usually it's more common to have separated controller instead of separated method

**Review by vojkovoga — CHANGES_REQUESTED** [review]

File: Controllers/BaseController.php (hunk 1 above)
Comment (zabukovnikmaj) [reply Controllers/BaseController.php]: The unnecessary code has been removed.

File: Controllers/BaseController.php (hunk 2 above)
Comment (zabukovnikmaj) [reply Controllers/BaseController.php:244]: Json helper method will be used here once merged.

File: config/routes.php (hunk 3 above)
Comment (zabukovnikmaj) [reply config/routes.php]: New class for API has been created

**Pull request merged**